document/FileFormat.mdk

```diff
@@ -2696,31 +2703,318 @@ The tool might provide a "related location" (&sect;[#result-relatedLocations]) w
+
+```
+f(7, m + n, "s", this, g(2));
+```
+
+then the value of `kind` would be `"call"`, the value of `target` might be `"f"` (or, for example, `"N.C.f"` if
+the function `f` occurred in class `C` in namespace `N`), the value of `parameters` would be
+
+```
+[ "7", "m + n", "\"s\"", "this", "g(2)" ]
+```
+
+and the value of `variables` might be
+
+```
+{ "m": "2", "n": "3" }
+```
+
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the function call:
+
+```
+int n = f();
+```
+
+then the value of `kind` would be `"callReturn"`, the value of `target` might be `"f"` (or, for example, `"N.C.f"` if
+the function `f` occurred in class `C` in namespace `N`), the value of `parameters` might be
+
+```
+[ "5" ]
+```
+
+and `variables` would be absent. 
+
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the declaration
+
+```
+int m = n + p;
+```
+
+then the value of `kind` would be `"declaration"`, the value of `target` would be `"m"`,
+the value of `parameters` might be
+
+```
+[ "5" ]
+```
+
+and the value of `variables` might be
+
+```
+{ "n": "2", "p": "3" }
+```
+
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the declaration
+
+```
+int m = n + p, q = k + r;
+```
+
+and if the tool creating the log wished to represent the declarations of both variables in the log file, then the tool would create
+two `annotatedCodeLocation` objects, each with `kind` set to `"declaration"`, and referring to the same source line.
+~
+
+~IsoNumberedExample
+In C++ or C#, if the source code contains the `return` statement
 
-If the value of the `kind` property (&sect;[#annotatedCodeLocation-kind]) of this `annotatedCodeLocation` object
-is `call`, then the `annotatedCodeLocation` object should contain a property named `callee` whose value is
-the fully qualified logical name of the function being called.
+```
+int f()
+{
+    ...
+    return m + n;
+}
+```
+
+then the value of `kind` would be `"functionExit"`, the value of `target` might be `"f"` (or, for example, `"N.C.f"` if
+the function `f` occurred in class `C` in namespace `N`), the value of `parameters` might be
 
-If the value of the `kind` property is anything other than `call`, the `callee` property shall be absent.
+```
+[ "5" ]
+```
 
-The format of the `callee` string shall be consistent with the programming language
-in which the function specified by that string was written.
+and the value of `variables` might be
+
+```
+{ "m": "2", "n": "3" }
+```
+~
 
 If the `run.logicalLocations` property (&sect;[#run-logicalLocations]) is present,
-the value of the `callee` property should be equal to the name of one of the properties on the `run.logicalLocations` object,
-with one exception, described in &sect;[#annotatedCodeLocation-calleeKey].
+and the the value of `kind` is `"call"`, then
+the value of the `target` property should be equal to the name of one of the properties on the `run.logicalLocations` object,
```

Comment:
>should be [](start = 35, length = 9)

should be means that we recommend this? it seems like a bit of overkill to encourage people to provide logical location breakdown for a called item. remember, these mostly exist to help locate things when source isn't available. this all points out one missed opportunity, to actually specify a source location for a logical item. imagine that, for a call, someone did specify a logical location (looked up by the fully qualified name). we could fix this by providing a targetLocation (a physicalLocation associated with a target). or we could add an optional physicalLocation with any logicalLocation. or we could ignore the gap.